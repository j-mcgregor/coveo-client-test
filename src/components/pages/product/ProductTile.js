import React from 'react';
import PropTypes from 'prop-types';
import { useRouteMatch, Link } from 'react-router-dom';
import S from '../../../static/styles';

const ProductTile = ({ product, type, cols }) => {
  const { url } = useRouteMatch();
  let item = {};
  if (product) {
    switch (type) {
      case 'beers':
        item = {
          id: product.id,
          name: product.name,
          location: `${product.city}, ${product.state}`,
          year: '',
          url: product.website_url
        };
        break;
      case 'wines':
        item = {
          id: product.wine_id,
          name: product.wine,
          location:
            product.regions &&
            product.regions.length &&
            `${product.regions[0]}, ${product.country}`,
          year: product.vintage,
          url: ''
        };
        break;
      default:
        break;
    }
  }
  const responsiveCols = cols === 2 ? 'col-2 col-md-4' : 'col-6 col-md-12';

  return (
    <S.Card className={`column ${responsiveCols}`}>
      <Link to={`${url}/${item.id}`}>
        <div className="card" style={{ height: '13rem', margin: '1rem 0' }}>
          <div className="card-header">
            <div className="card-title h6">{item.name}</div>
            <small className="card-subtitle">{item.location}</small>
          </div>
          <div className="card-body" />
          <div className="card-footer columns">
            <div className="column col">
              <small className="card-subtitle text-gray">{item.year}</small>
            </div>
          </div>
        </div>
      </Link>
    </S.Card>
  );
};

ProductTile.propTypes = {
  product: PropTypes.shape({}),
  type: PropTypes.string,
  cols: PropTypes.number
};

ProductTile.defaultProps = {
  product: {},
  type: '',
  cols: 2
};

export default ProductTile;
