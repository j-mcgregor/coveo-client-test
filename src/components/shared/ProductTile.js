import PropTypes from 'prop-types';
import React from 'react';
import { Link, useRouteMatch } from 'react-router-dom';
import S from '../../static/styles';
import { createItem, makeResponsiveColumns } from '../../utils/helpers';

const ProductTile = ({ product, type, cols, selectProduct, truncate }) => {
  const { url } = useRouteMatch();
  let item = {};
  if (product) {
    item = createItem(type, product, truncate);
  }

  const responsiveCols = makeResponsiveColumns(cols);

  const handleClick = () => selectProduct(product);

  const content = (
    <div className="card">
      <div className="card-header">
        <div className="card-title h6">{item.name}</div>
        <small className="card-subtitle">{item.location || item.price}</small>
      </div>
      {item.thumbnail && (
        <div className="card-body">
          <img src={item.thumbnail} alt="" />
        </div>
      )}
      {!item.thumbnail && (
        <div className="card-footer columns">
          <div className="column col">
            <small className="card-subtitle text-gray">{item.year}</small>
          </div>
        </div>
      )}
    </div>
  );

  return (
    <S.Card className={`column ${responsiveCols}`}>
      {type === 'generic' ? (
        <div onClick={handleClick} role="presentation">
          {content}
        </div>
      ) : (
        <Link onClick={handleClick} to={`${url}/${item.id}`}>
          {content}
        </Link>
      )}
    </S.Card>
  );
};

ProductTile.propTypes = {
  product: PropTypes.shape({}).isRequired,
  type: PropTypes.string,
  cols: PropTypes.number,
  selectProduct: PropTypes.func,
  truncate: PropTypes.bool
};

ProductTile.defaultProps = {
  type: '',
  cols: 2,
  selectProduct: () => {},
  truncate: false
};

export default ProductTile;
