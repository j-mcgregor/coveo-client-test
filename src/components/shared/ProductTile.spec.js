import React from 'react';
import { shallow } from 'enzyme';
import ProductTile from './ProductTile';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
  useRouteMatch: () => ({ url: '/test' })
}));

describe('ProductTile', () => {
  it('should render', () => {
    const props = {
      product: {},
      type: '',
      cols: null,
      selectProduct: () => {}
    };

    const wrapper = shallow(<ProductTile {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
