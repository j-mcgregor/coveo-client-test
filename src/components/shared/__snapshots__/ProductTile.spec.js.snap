// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTile should render 1`] = `ShallowWrapper {}`;
